db/change_cache.go

```diff
@@ -473,9 +473,16 @@ func (c *changeCache) DocChanged(event sgbucket.FeedEvent) {
 
 	if len(syncData.RecentSequences) > 0 {
 		nextSequence := c.getNextSequence()
+		oldestSkipped := c.getOldestSkippedSequence(ctx)
 
 		for _, seq := range syncData.RecentSequences {
-			if seq >= nextSequence && seq < currentSequence {
+			// seq < currentSequence means the sequence is not the latest allocated to this document
```

Comment:
Functionally it looks like what you're doing here is:
1. Ignore recent sequences that aren't less than the doc's current sequence (existing corner case handling)
2. For any others, let n be the lowest sequence of [oldest skipped, nextSequence].  Send any recent sequences through process entry if they are greater than or equal to n.

This lets process entry do the work of determining whether they are skipped, pending, or already processed.

If that's accurate, I think you can reorganize the code to compute n outside the loop, which would make the the logic inside the loop easier to reason about.